__init__.py

```diff
@@ -11,1810 +11,950 @@
-            timezone = pytz.timezone(self.location["timezone"]["code"])
-            return when.astimezone(timezone)
-
-    def __to_time_period(self, when):
-        # Translate a specific time '9am' to period of the day 'morning'
-        hour = when.time().hour
-        period = None
-        if hour >= 1 and hour < 5:
-            period = "early morning"
-        if hour >= 5 and hour < 12:
-            period = "morning"
-        if hour >= 12 and hour < 17:
-            period = "afternoon"
-        if hour >= 17 and hour < 20:
-            period = "evening"
-        if hour >= 20 or hour < 1:
-            period = "overnight"
-        if period is None:
-            self.log.error("Unable to parse time as a period of day")
-        return period
-
-    # Suggestion TODO: Add a parameter to extract_datetime to add a default Timezone
-    def __extract_datetime(self, text, anchorDate=None, lang=None, default_time=None):
-        # Change timezone returned by extract_datetime from Local to UTC
-        extracted_dt = extract_datetime(text, anchorDate, lang, default_time)
-        if extracted_dt is None:
-            # allow calls to unpack values even if None returned.
-            return (None, None)
-        when, text = extracted_dt
-        return to_utc(when), text
-
-    def __translate(self, condition, future=False, data=None):
-        # behaviour of method dialog_renderer.render(...) has changed - instead
-        # of exception when given template is not found now simply the
-        # templatename is returned!?!
-        if (future and
-                (condition + ".future") in self.dialog_renderer.templates):
-            return self.translate(condition + ".future", data)
-        if condition in self.dialog_renderer.templates:
-            return self.translate(condition, data)
+        if intent_data.location is None:
+            latitude = self.weather_config.latitude
+            longitude = self.weather_config.longitude
         else:
-            return condition
+            latitude = intent_data.geolocation["latitude"]
+            longitude = intent_data.geolocation["longitude"]
 
-    def __nice_time(self, dt, lang="en-us", speech=True, use_24hour=False,
-                    use_ampm=False):
-        # compatibility wrapper for nice_time
-        nt_supported_languages = ['en', 'es', 'it', 'fr', 'de',
-                                  'hu', 'nl', 'da']
-        if not (lang[0:2] in nt_supported_languages):
-            lang = "en-us"
-        return nice_time(dt, lang, speech, use_24hour, use_ampm)
+        return latitude, longitude
+
+    def _speak_weather(self, dialog: WeatherDialog):
+        """Instruct device to speak the contents of the specified dialog.
+
+        :param dialog: the dialog that will be spoken
+        """
+        self.log.info("Speaking dialog: " + dialog.name)
+        self.speak_dialog(dialog.name, dialog.data)
+        mycroft.audio.wait_while_speaking()
```

Comment:
This can be simplified to

```python
self.speak_dialog(dialog.name, dialog.data, wait=True)
```

Reply:
I got rid of the wait command and passed `wait=True` to `self.speak_dialog`, but I want to keep the log message.